src/CMakeLists.txt

```
@@ -61,12 +61,16 @@ if(nfd_PLATFORM STREQUAL PLATFORM_LINUX)
 endif()
 
 if(nfd_PLATFORM STREQUAL PLATFORM_MACOS)
-    if(UNIFORMTYPEIDENTIFIERS_LIBRARY)
-      target_link_libraries(${TARGET_NAME}
-        PRIVATE ${APPKIT_LIBRARY} ${UNIFORMTYPEIDENTIFIERS_LIBRARY})
+    if(NFD_USE_ALLOWEDCONTENTTYPES AND CMAKE_OSX_DEPLOYMENT_TARGET VERSION_GREATER_EQUAL "11.0")
+      if (NOT UNIFORMTYPEIDENTIFIERS_LIBRARY)
+        message(FATAL_ERROR "UniformTypeIdentifiers Framework is not available!")
+      endif ()
+      
+      target_link_libraries(${TARGET_NAME} PRIVATE ${APPKIT_LIBRARY} ${UNIFORMTYPEIDENTIFIERS_LIBRARY})
+      target_compile_definitions(${TARGET_NAME} PRIVATE NFD_MACOS_ALLOWEDCONTENTTYPES=1)
     else()
-      target_link_libraries(${TARGET_NAME}
-        PRIVATE ${APPKIT_LIBRARY})
+      target_link_libraries(${TARGET_NAME} PRIVATE ${APPKIT_LIBRARY})
+      target_compile_definitions(${TARGET_NAME} PRIVATE NFD_MACOS_ALLOWEDCONTENTTYPES=0)
```

I asked the question on StackOverflow, but until there is a reasonable answer, I think we should just add the option as you suggest, so that users can manually configure it, but don't change the whether the source file is compiled with allowedContentTypes or not.

Let's change this back:
```suggestion
    if(NFD_USE_ALLOWEDCONTENTTYPES AND UNIFORMTYPEIDENTIFIERS_LIBRARY)
      target_link_libraries(${TARGET_NAME}
        PRIVATE ${APPKIT_LIBRARY} ${UNIFORMTYPEIDENTIFIERS_LIBRARY})
    else()
      target_link_libraries(${TARGET_NAME}
        PRIVATE ${APPKIT_LIBRARY})
    endif()
```

---

So the `NFD_MACOS_ALLOWEDCONTENTTYPES` shouldn't be set anymore either and needs to be set by the user in their cmake script as well?

---

No. Setting `NFD_MACOS_ALLOWEDCONTENTTYPES` to ON (the default) should give us the original behaviour, so it is not a breaking change. Setting `NFD_MACOS_ALLOWEDCONTENTTYPES` to OFF prevents linking with the UniformTypeIdentifiers library, and you can use that to fix your build for ImHex.

---

Oh, I see what you mean now.